javascripts/discourse/initializers/discourse-image-annotator.js.es6

```diff
@@ -28,66 +60,111 @@ export default {
       const matches = editor.match(regex);
 
       oldImgUrl = matches[imageIndex];
+      // if (validation) {
+      img.src = loadingDataUri;
 
-      //Exec after marker's toolbar checkmark is clicked
-      mark.settings.displayMode = 'popup';
-      mark.addRenderEventListener((dataUrl) => {
-        if (mark.getState().markers.length > 0) {
-          img.src = loadingDataUri;
-
-          //generating dataurl into blob
-          dataUrl = dataUrl.split(",").pop();
-          const byteChar = atob(dataUrl);
-          const byteNum = new Array(byteChar.length);
-
-          for (let i = 0; i < byteChar.length; i++) {
-            byteNum[i] = byteChar.charCodeAt(i);
-          }
-          const byteArr = new Uint8Array(byteNum);
-          const blob = new Blob([byteArr], { type: "img/png" });
-
-          // upload
-          const data = new FormData();
-          data.append(
-            "authenticity_token",
-            document.querySelector('meta[name="csrf-token"]').content
-          );
-          data.append("file", blob, "png");
-          data.append("type", "composer");
-
-          fetch(getURL("/uploads.json"), {
-            method: "POST",
-            body: data,
-          })
-            .then((response) => response.json())
-            .then((upload) => {
-              //Replace old image url with modified image url in composer
-              let newImgUrl = upload.short_url;
-              
-              appEvents.trigger(
-                "composer:replace-text",
-                oldImgUrl,
-                newImgUrl
-              );
-            });
-          }
-      });
-      mark.show();
+      //generating dataurl into blob
+      dataUrl = dataUrl.split(",").pop();
+      const byteChar = atob(dataUrl);
+      const byteNum = new Array(byteChar.length);
+
+      for (let i = 0; i < byteChar.length; i++) {
+        byteNum[i] = byteChar.charCodeAt(i);
+      }
+      const byteArr = new Uint8Array(byteNum);
+      const blob = new Blob([byteArr], { type: "img/png" });
+
+      // upload
+      const data = new FormData();
+      data.append(
+        "authenticity_token",
+        document.querySelector('meta[name="csrf-token"]').content
+      );
+      data.append("file", blob, "png");
+      data.append("type", "composer");
+
+      fetch(getURL("/uploads.json"), {
+        method: "POST",
+        body: data,
+      })
+        .then((response) => response.json())
+        .then((upload) => {
+          //Replace old image url with modified image url in composer
+          let newImgUrl = upload.short_url;
+
+          appEvents.trigger("composer:replace-text", oldImgUrl, newImgUrl);
```

Comment:
We probably don't need the variable `newImgUrl` here. Lets just pass `upload.short_url` into the appEvent trigger